File: CS-Examples/CS-Examples/24_Workbook/SetTheme.cs
﻿using Spire.Xls;
using System;
using System.Drawing;
using System.Windows.Forms;

namespace SetTheme
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a workbook
            Workbook srcWorkbook = new Workbook();
            //Load an excel file
            srcWorkbook.LoadFromFile(@"..\..\..\..\..\..\Data\SetTheme.xlsx");
            Worksheet srcWorksheet = srcWorkbook.Worksheets[0];

            Workbook workbook = new Workbook();
            workbook.Worksheets.Clear();
            workbook.Worksheets.AddCopy(srcWorksheet);

            //1. Copy the theme of the workbook
            //workbook.CopyTheme(srcWorkbook);

            //2. Set a certain type of color of the default theme in the workbook
            workbook.SetThemeColor(ThemeColorType.Dk1, Color.SkyBlue);

            String result = "SetTheme_result.xlsx";
            //Save to file
            workbook.SaveToFile(result, ExcelVersion.Version2010);
            //View the document
            FileViewer(result);
          
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

        private void Form1_Load(object sender, EventArgs e)
        {

        }
    }
}

File: VB-Examples/24_Workbook/SetTheme.vb
Imports Spire.Xls

Namespace SetTheme
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object to hold the source workbook
            Dim srcWorkbook As New Workbook()

            ' Load an existing Excel file into the source workbook
            srcWorkbook.LoadFromFile("..\..\..\..\..\..\Data\SetTheme.xlsx")

            ' Get the first worksheet from the source workbook
            Dim srcWorksheet As Worksheet = srcWorkbook.Worksheets(0)

            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Clear any existing worksheets in the workbook
            workbook.Worksheets.Clear()

            ' Add a copy of the source worksheet to the workbook
            workbook.Worksheets.AddCopy(srcWorksheet)

            ' Copy the theme from the source workbook to the current workbook (commented out)
            ' workbook.CopyTheme(srcWorkbook)

            ' Set the theme color for the workbook's theme
            workbook.SetThemeColor(ThemeColorType.Dk1, Color.SkyBlue)

            ' Specify the output file name
            Dim result As String = "SetTheme_result.xlsx"

            ' Save the workbook to the specified output file in Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(result)

		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load

		End Sub
	End Class
End Namespace
